### PR TITLE
AWS Sagemaker Clone Training Job

### PR DESCRIPTION
**Pull request by sgokaram-saagie**
Sagemaker Clone Training Job

### CODE REVIEW

File: technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+            }
+          },
+          CompressionType: 'None',
+          InputMode: 'File'
+        }
+    ],
+    Tags: data.Tags
+    }
+
+    const clone=await sagemaker.createTrainingJob( params = jparams ).promise();
```
Comment (yoannfleurydev) [inline technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js:78]:
```suggestion
    const clone=await sagemaker.createTrainingJob(jparams).promise();
```

No need to reassign to another variable.

File: technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    console.log('Get Job Details',  job.featuresValues.trainingjobs.label)
+    const data = await sagemaker.describeTrainingJob( params = {
+      TrainingJobName: job.featuresValues.trainingjobs.label
+    }).promise();
+
+    console.log("Retrieved Template Job Details");
+    console.log(data);
+
+    console.log("Create New Job based on template parameters");
+    var jparams = {
```
Comment (yoannfleurydev) [inline technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js:40]:
```suggestion
    const jparams = {
```

https://github.com/saagie/sdk/wiki/Best-Practices#var-let-and-const-

File: technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+ */
+exports.stop = async ({ job, instance }) => {
+  try {
+    console.log('STOP INSTANCE:', instance);
+    AWS.config.update({credentials: { accessKeyId : job.featuresValues.endpoint.aws_access_key_id, secretAccessKey:  job.featuresValues.endpoint.aws_secret_access_key}});
+    AWS.config.update({region: job.featuresValues.endpoint.region});
+    
+    const sagemaker = new AWS.SageMaker({apiVersion: '2017-07-24'});
+
+    await sagemaker.stopTrainingJob(params =  {
```
Comment (yoannfleurydev) [inline technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js:104]:
```suggestion
    await sagemaker.stopTrainingJob({
```

File: technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+ */
+exports.getStatus = async ({ job, instance }) => {
+  try {
+    console.log('GET STATUS INSTANCE:', instance);
+    AWS.config.update({credentials: { accessKeyId : job.featuresValues.endpoint.aws_access_key_id, secretAccessKey:  job.featuresValues.endpoint.aws_secret_access_key}});
+    AWS.config.update({region: job.featuresValues.endpoint.region});
+    
+    const sagemaker = new AWS.SageMaker({apiVersion: '2017-07-24'});
+    console.log('Get Job Details',  job.featuresValues.trainingjobs.label)
+    const data = await sagemaker.describeTrainingJob( params = {
```
Comment (yoannfleurydev) [inline technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js:128]:
```suggestion
    const data = await sagemaker.describeTrainingJob({
```

File: technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+ */
+exports.getLogs = async ({ job, instance }) => {
+  try {
+    console.log('GET LOG INSTANCE:', instance);
+    AWS.config.update({credentials: { accessKeyId : job.featuresValues.endpoint.aws_access_key_id, secretAccessKey:  job.featuresValues.endpoint.aws_secret_access_key}});
+    AWS.config.update({region: job.featuresValues.endpoint.region});
+    
+    const sagemaker = new AWS.SageMaker({apiVersion: '2017-07-24'});
+    console.log('Get Job Details',  job.featuresValues.trainingjobs.label)
+    const data = await sagemaker.describeTrainingJob( params = {
```
Comment (yoannfleurydev) [inline technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js:163]:
```suggestion
    const data = await sagemaker.describeTrainingJob({
```

File: technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+      TrainingJobName: job.featuresValues.trainingjobname
+    }).promise();
+
+    const cloudwatchlogs = new AWS.CloudWatchLogs({apiVersion: '2014-03-28'});
+
+    const logs = await cloudwatchlogs.filterLogEvents(params = {
```
Comment (yoannfleurydev) [inline technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js:169]:
```suggestion
    const logs = await cloudwatchlogs.filterLogEvents({
```

File: technologies/job/aws-sagemaker/clone-trainingjob/jobForm.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+};
+
+
+exports.getJobOutput = async ({ featuresValues }) => {
+  try {
+    AWS.config.update({credentials: { accessKeyId : featuresValues.endpoint.aws_access_key_id, secretAccessKey:  featuresValues.endpoint.aws_secret_access_key}});
+    AWS.config.update({region: featuresValues.endpoint.region});
+    const sagemaker = new AWS.SageMaker();
+
+    const data = await sagemaker.describeTrainingJob( params = {
```
Comment (yoannfleurydev) [inline technologies/job/aws-sagemaker/clone-trainingjob/jobForm.js:60]:
```suggestion
    const data = await sagemaker.describeTrainingJob({
```

File: technologies/job/aws-sagemaker/clone-trainingjob/jobForm.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+};
+
+
+exports.getJobInput = async ({ featuresValues }) => {
+  try {
+    AWS.config.update({credentials: { accessKeyId : featuresValues.endpoint.aws_access_key_id, secretAccessKey:  featuresValues.endpoint.aws_secret_access_key}});
+    AWS.config.update({region: featuresValues.endpoint.region});
+    const sagemaker = new AWS.SageMaker();
+
+    const data = await sagemaker.describeTrainingJob( params = {
```
Comment (yoannfleurydev) [inline technologies/job/aws-sagemaker/clone-trainingjob/jobForm.js:39]:
```suggestion
    const data = await sagemaker.describeTrainingJob({
```

File: technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js (hunk 1 above)
Comment (sgokaram-saagie) [reply technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js:78]: Done

File: technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+          CompressionType: 'None',
+          InputMode: 'File'
+        }
+    ],
+    Tags: data.Tags
+    }
+
+    const clone=await sagemaker.createTrainingJob(jparams).promise();
+
+    console.log(clone);
```
Comment (youenchene) [inline technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js:80]: To clean/remove

File: technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+
+    if ( job.featuresValues.trainingjoboutput == null || job.featuresValues.trainingjoboutput.trim() === '' ) {
+          return Response.error('Training Job Output S3 location is required.', new Error("Validation Error"));
+    }
+
+    AWS.config.update({credentials: { accessKeyId : job.featuresValues.endpoint.aws_access_key_id, secretAccessKey:  job.featuresValues.endpoint.aws_secret_access_key}});
+    AWS.config.update({region: job.featuresValues.endpoint.region});
+    
+    const sagemaker = new AWS.SageMaker({apiVersion: '2017-07-24'});
+    console.log('Get Job Details',  job.featuresValues.trainingjobs.label)
```
Comment (youenchene) [inline technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js:31]: To clean/remove.

File: technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+    const data = await sagemaker.describeTrainingJob( params = {
+      TrainingJobName: job.featuresValues.trainingjobs.label
+    }).promise();
+
+    console.log("Retrieved Template Job Details");
+    console.log(data);
+
+    console.log("Create New Job based on template parameters");
```
Comment (youenchene) [inline technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js:39]: To clean/remove

File: technologies/job/aws-sagemaker/clone-trainingjob/jobForm.js [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+  try {
+    AWS.config.update({credentials: { accessKeyId : featuresValues.endpoint.aws_access_key_id, secretAccessKey:  featuresValues.endpoint.aws_secret_access_key}});
+    AWS.config.update({region: featuresValues.endpoint.region});
+    const sagemaker = new AWS.SageMaker();
+
+    const data = await sagemaker.describeTrainingJob({
+          TrainingJobName: featuresValues.trainingjobs.label
+        }).promise();
+
+    console.log(data.InputDataConfig);
```
Comment (youenchene) [inline technologies/job/aws-sagemaker/clone-trainingjob/jobForm.js:43]: To clean/remove

File: technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js [hunk 13]
```diff
@@ -0,0 +1,182 @@
...
+ * @param {Object} params.instance - Contains instance data including the payload returned in the start function.
+ */
+exports.getStatus = async ({ job, instance }) => {
+  try {
+    console.log('GET STATUS INSTANCE:', instance);
+    AWS.config.update({credentials: { accessKeyId : job.featuresValues.endpoint.aws_access_key_id, secretAccessKey:  job.featuresValues.endpoint.aws_secret_access_key}});
+    AWS.config.update({region: job.featuresValues.endpoint.region});
+    
+    const sagemaker = new AWS.SageMaker({apiVersion: '2017-07-24'});
+    console.log('Get Job Details',  job.featuresValues.trainingjobs.label)
```
Comment (youenchene) [inline technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js:127]: To clean/remove

File: technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js [hunk 14]
```diff
@@ -0,0 +1,182 @@
...
+    const data = await sagemaker.describeTrainingJob({
+      TrainingJobName: job.featuresValues.trainingjobname
+    }).promise();
+
+    console.log(data)
```
Comment (youenchene) [inline technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js:132]: To clean/remove

**Pull request merged**